llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -485,6 +502,72 @@ class MCCASReader {
   const MCSchema &Schema;
 };
 
+class InMemoryCASDWARFObject : public DWARFObject {
+  ArrayRef<char> DebugAbbrevSection;
+  bool IsLittleEndian;
+
+public:
+  InMemoryCASDWARFObject(ArrayRef<char> AbbrevContents, bool IsLittleEndian)
+      : DebugAbbrevSection(AbbrevContents), IsLittleEndian(IsLittleEndian) {}
+  bool isLittleEndian() const override { return IsLittleEndian; }
+
+  StringRef getAbbrevSection() const override {
+    return toStringRef(DebugAbbrevSection);
+  }
+
+  Optional<RelocAddrEntry> find(const DWARFSection &Sec,
+                                uint64_t Pos) const override {
+    return {};
+  }
+
+  struct SplitUpCUData {
+    SmallVector<char> DebugInfoRefData;
+    SmallVector<char> FixupData;
+  };
+  /// Identify the Forms specified in /p FormsToFixup in the compile unit and
+  /// zero them out, while also saving them as Addends to the debug_info
+  /// section. Addends are stored per section and DW_AT_stmt_list offsets are
+  /// stored per compile unit, so pass a /p CUOffset to store the
+  /// DW_AT_stmt_list offset in the section rather than in the compile unit.
```

Comment:
This comment is still reflecting the old approach, not what the new PR is doing!

Reply:
Thanks for pointing it out!